pyscriptjs/src/python/pyscript/_deprecated_globals.py

```diff
@@ -0,0 +1,160 @@
+        """
+        NOTE: this is overridden by unit tests
+        """
+        showWarning(message, "html")  # noqa: F821
+
+    def _show_warning_maybe(self):
+        if self.__warning_already_shown:
+            return
+        self._show_warning(self.__message)
+        self.__warning_already_shown = True
+
+    def __getattr__(self, attr):
+        self._show_warning_maybe()
+        return getattr(self.__obj, attr)
+
+    def __call__(self, *args, **kwargs):
+        self._show_warning_maybe()
+        return self.__obj(*args, **kwargs)
+
+    def __iter__(self):
+        self._show_warning_maybe()
+        return iter(self.__obj)
+
+    def __getitem__(self, key):
+        self._show_warning_maybe()
+        return self.__obj[key]
+
+    def __setitem__(self, key, value):
+        self._show_warning_maybe()
+        self.__obj[key] = value
+
+
+def deprecate(name, obj, instead):
+    message = f"Direct usage of <code>{name}</code> is deprecated. " + instead
+    main_dict[name] = DeprecatedGlobal(name, obj, message)
+
+
+def deprecated_pyscript_globals():
+    # function/classes defined in pyscript.py ===> pyscript.XXX
+    from . import __dict__ as public_pyscript
+
+    pyscript_names = [
+        "PyItemTemplate",
+        "PyListTemplate",
+        "PyWidgetTheme",
+        "add_classes",
+        "create",
+        "loop",
+    ]
+    for name in pyscript_names:
+        deprecate(
+            name,
+            public_pyscript[name],
+            f"Please use <code>pyscript.{name}</code> instead.",
+        )
+
+
+def deprecated_mime_globals():
+    from ._mime import __dict__ as mime_ns
+
+    # these are names that used to leak in the globals but they are just
+    # implementation details. People should not use them.
+    private_names = [
+        "eval_formatter",
+        "format_mime",
+        "identity",
+        "render_image",
+        "MIME_RENDERERS",
+        "MIME_METHODS",
+    ]
+    for name in private_names:
+        obj = mime_ns[name]
+        message = (
+            f"<code>{name}</code> is deprecated. "
+            "This is a private implementation detail of pyscript. "
+            "You should not use it."
+        )
+        main_dict[name] = DeprecatedGlobal(name, obj, message)
+
+
+def deprecate_stdlib_modules():
+    # stdlib modules ===> import XXX
+    stdlib_names = [
+        "asyncio",
+        "base64",
+        "io",
+        "sys",
+        "time",
+        "datetime",
+        "pyodide",
+        "micropip",
+    ]
+    for name in stdlib_names:
+        obj = __import__(name)
+        deprecate(name, obj, f"Please use <code>import {name}</code> instead.")
+
+
+def install_deprecated_globals_2022_12_1():
+    """
+    Install into the given namespace all the globals which have been
+    deprecated since the 2022.12.1 release. Eventually they should be killed.
+    """
+    deprecated_pyscript_globals()
+    deprecated_mime_globals()
+    deprecate_stdlib_modules()
+
```

Comment:
as the comment says, we should kill these completely, since we completed the deprecation cycle.
I wanted to open a PR just for doing that, but since you are touching the code anyway in this PR, maybe we should just kill it altogether (but keep the `DeprecatedGlobal` class, as it might still be useful in the future)

Reply:
Can't we just reintroduce `DeprecatedGlobal` from the git history if we need it again?

Reply:
in my experience this never works well in practice, because people tend to forget about its existence, and even when they remember if they try to revive it from git history it's probably half-broken.
But +1 for doing this is you really want. In that case, please write comment with a link to a revision which actually contains the code, so that it's easy to retrieve.